src/components/NewToast.tsx

```diff
@@ -0,0 +1,111 @@
+  onClose: () => void;
+  toastIsReady: () => void;
+}
+export default function NewToast({ onClose, toastIsReady }: NewToastProps) {
+  const [toastHeader, setToastHeader] = useState("");
+  const [toastStatus, setToastStatus] = useState("");
+  const [taskRunning, setTaskRunning] = useState(true);
+
+  useEffect(() => {
+    const eventSource = new EventSource("/api/events"); // issue GET to open connection to the SSE server
+    console.log(
+      `NewToast has opened a connection: ${JSON.stringify(eventSource)}`,
+    );
+
+    eventSource.onmessage = (event: MessageEvent) => {
+      try {
+        const data: { status: string; type: string; header: string } =
+          JSON.parse(event.data);
+
+        if (data.type === "ready") {
+          console.log(`received ready message`);
+          toastIsReady();
+          return;
+        }
+        if (data.type === "update") {
+          // just an update
+          setToastHeader(data.header);
+          setToastStatus(data.status);
+        }
+
+        if (data.type == "done") {
+          // we're done (complete, error, canceled)
+          setToastHeader(data.header);
+          setToastStatus(data.status);
+          eventSource.close();
+          setTimeout(() => {
+            setTaskRunning(false);
+            onClose();
+          }, 3000);
+        }
+      } catch (error) {
+        console.error("Error parsing SSE message:", error);
+      }
+    };
+
+    eventSource.onerror = (error: Event) => {
+      console.error("SSE error:", error);
+      eventSource.close();
+    };
+
+    // Handle browser reload/unload
+    const handleUnload = () => {
+      eventSource.close(); // cleanly closes connection
+    };
+
+    window.addEventListener("beforeunload", handleUnload);
+    window.addEventListener("unload", handleUnload);
+
+    return () => {
+      eventSource.close();
+      window.removeEventListener("beforeunload", handleUnload);
+      window.removeEventListener("unload", handleUnload);
+    };
+  }, []);
+
+  const handleCancel = async () => {
+    // tell the server to stop doing work
+    await fetch("/api/start-task?action=stop", { method: "POST" });
+    setToastStatus("Task Canceled ❌");
+    setToastHeader("Canceled");
+    setTaskRunning(false);
+    setTimeout(() => onClose(), 3000);
+  };
+
+  return (
+    <Toast.Provider swipeDirection="right">
+      <Toast.Root
+        className="fixed bottom-[10px] right-[5px] w-96 bg-gray-800 text-white p-4 rounded shadow-md flex justify-between items-center"
+        duration={Infinity} // Keeps open until manually closed
+      >
+        <div>
+          <Toast.Title className="font-bold">{toastHeader}</Toast.Title>
+          {/* Convert \n into actual <br /> elements */}
+          <Toast.Description className="text-sm text-gray-700 leading-relaxed">
+            {toastStatus.split("\n").map((line, index) => (
+              <span key={index}>
+                {line}
+                <br />
+              </span>
```

Comment:
Is the br really necessary? Isn't it enough just to have this in something like 

```
<div className="flex flex-col">
<div>
{line}
<div>
</div>
```
instead?

Reply:
Seems OK